src/tokens/tokens.service.ts

```diff
@@ -432,7 +444,19 @@ export class TokensService {
   }
 
   async supportsData(address: string, type: TokenType) {
-    const iid = type === TokenType.FUNGIBLE ? ERC20WithDataIID : ERC721WithDataIID;
+    const nftIID = await this.supportsNFTUri(address, false) ? ERC721WithDataUriIID : ERC20WithDataIID
```

Comment:
If `supportsNFTURI` returns true, doesn't that mean it also supports data? Seems like we can short-circuit and return much earlier.